model_rnn/dataset.py

```diff
@@ -0,0 +1,52 @@
+import pandas as pd
+import re
+import string
+import unicodedata
+import torch
+import torch.nn as nn
+import torch.utils.data as data
+from .vectorize import make_input_vect
+from .config import DATA_FILEPATH
+from .config import DATA_ALL_CHARS, BOS, EOS, DATA_CHAR_TO_IX
+
+
+class Dataset(data.Dataset):
+    def __init__(self):
+        self.titles = [self._normalize_title(x) for x in self.read_data()]
+        self.preprocess()
+
+    def __iter__(self):
+        return iter(self.titles)
+    
+    def __len__(self):
+        return len(self.titles)
+    
+    def __getitem__(self, index):
+        return make_input_vect(self.titles[index])
+
+    def read_data(self) -> list[str]:
+        return pd.read_csv(DATA_FILEPATH)['PRODUCT_NAME']\
+            .dropna().astype(str).tolist()
+
+    def _normalize_title(self, s:str) -> str:
+        s = []
+        for char in unicodedata.normalize('NFD', string.strip()):
+            if unicodedata.category(char) != 'Mn' and char in DATA_ALL_CHARS:
+                s.append(char)
+        s = ''.join(s)
+        s = re.sub(r' +', ' ', s)
+        return BOS + s + EOS
```

Comment:
Need to move this out to preprocessor.py file, this will be used during inference too.